src/SitemapGenerator.php

```diff
@@ -150,8 +185,12 @@ protected function getCrawlObserver(): Observer
         $performAfterUrlHasBeenCrawled = function (UriInterface $crawlerUrl, ResponseInterface $response = null) {
             $sitemapUrl = ($this->hasCrawled)(Url::create((string) $crawlerUrl), $response);
 
+            if ($this->chunk and count($this->sitemaps->first()->getTags()) >= $this->chunk) {
```

Comment:
Extract this to a function with an expressive name such as `shouldAddSitemap` or something better.